### PR TITLE
[CSL-2290] JS Client SDK: Parameter Consistency

### PR DESCRIPTION
**Pull request by esezen**

### CODE REVIEW

File: src/modules/tracker.js [hunk 1]
```diff
@@ -218,26 +218,26 @@ class Tracker {
         name,
         item_id,
         customer_id,
+        customerId = customer_id,
```
Comment (stanlp1) [inline src/modules/tracker.js:221]: Do we need the customerId here?

Doesn't seem like we had it before so no one uses it probably?

File: src/modules/tracker.js [hunk 2]
```diff
@@ -579,32 +598,39 @@ class Tracker {
...
     // Ensure term is provided (required)
-    if (search_term && typeof search_term === 'string') {
+    if (searchTerm && typeof searchTerm === 'string') {
       // Ensure parameters are provided (required)
       if (parameters && typeof parameters === 'object' && !Array.isArray(parameters)) {
         const baseUrl = `${this.behavioralV2Url}search_result_load?`;
         const {
           num_results,
-          result_count = num_results,
+          numResults = num_results,
```
Comment (stanlp1) [inline src/modules/tracker.js:618]: Similarly here, do we need the numResults?

**Review by stanlp1 — APPROVED** [review]
Minor comment but this is looking good to me.

**Review by mocca102 — APPROVED** [review]
That's humongous Thanks for taking on that. LGTM!

Are we only doing tracker module?

File: src/modules/tracker.js (hunk 2 above)
Comment (esezen) [reply src/modules/tracker.js:618]: Same as above

**Comment by esezen** [comment]
> Are we only doing tracker module?

I believe that's the only one that uses camel case in this repo. Let me know if you find any other function that uses camel case

File: spec/src/modules/tracker.js [hunk 3]
```diff
@@ -742,6 +742,102 @@ describe(`ConstructorIO - Tracker${bundledDescriptionSuffix}`, () => {
...
+
+      tracker.on('error', (error) => {
+        done(error);
+      });
+
+      // eslint-disable-next-line max-len
+      expect(tracker.trackAutocompleteSelectV2(term, snakeCaseParameters)).to.equal(true);
+    });
+
+    it('Backwards Compatibility - Should respond with a valid response when snake cased parameters are provided', (done) => {
```
Comment (jjl014) [inline spec/src/modules/tracker.js:799]: I noticed that this backwards compatibility test and the one above it says `...when snake cased parameters are provided`, but the other backwards compatibility tests below says `...when required parameters are provided`.

Should those also say `...when snake cased parameters are provided`?

**Review by jjl014 — COMMENTED** [review]
Yowza! This was a big boi! Thank you for being so thorough and updating everything as well as making sure things are backwards compatible!

It's looking pretty great to me! I did leave a couple of minor comments here and there.

Also, I noticed in some functions we specify a default `section` (i.e. `section="Products"`) in the doc comments as well as defaulting the variable to `Products` when de-structuring. Should we do this for all functions to be consistent?

Thanks!

File: src/modules/tracker.js [hunk 4]
```diff
@@ -1207,34 +1266,35 @@ class Tracker {
...
-   *         result_id: '019927c2-f955-4020-8b8d-6b21b93cb5a2',
-   *         result_count: 22,
-   *         result_page: 2,
-   *         result_position_on_page: 2,
-   *         num_results_per_page: 12,
-   *         pod_id: '019927c2-f955-4020',
-   *         strategy_id: 'complimentary',
-   *         item_name: 'Socks',
-   *         item_id: 'KMH876',
+   *         resultId: '019927c2-f955-4020-8b8d-6b21b93cb5a2',
```
[the remark this answers is not in the document]
Comment (esezen) [reply src/modules/tracker.js]: Good catch! Fixed in [7861019](https://github.com/Constructor-io/constructorio-client-javascript/pull/262/commits/78610199469b8e551965bdda5ca129dc0172c2eb)

File: src/modules/tracker.js [hunk 5]
```diff
@@ -440,32 +449,35 @@ class Tracker {
...
         const baseUrl = `${this.behavioralV2Url}search_submit?`;
         const {
           original_query,
-          user_input = original_query,
+          originalQuery = original_query,
+          user_input,
+          userInput = originalQuery || user_input,
           group_id,
+          groupId = group_id,
           section,
```
[the remark this answers is not in the document]
Comment (esezen) [reply src/modules/tracker.js:471]: Good catch! I don't believe this is related to this PR but I checked the parameters and it looks like V2 takes in section. So kept the parameter and added documentation in [7861019](https://github.com/Constructor-io/constructorio-client-javascript/pull/262/commits/78610199469b8e551965bdda5ca129dc0172c2eb)

File: src/modules/tracker.js [hunk 6]
```diff
@@ -579,32 +598,39 @@ class Tracker {
...
           result_id,
+          resultId = result_id,
           sort_order,
+          sortOrder = sort_order,
           sort_by,
+          sortBy = sort_by,
           selected_filters,
+          selectedFilters = selected_filters,
           url,
           section,
```
[the remark this answers is not in the document]
Comment (esezen) [reply src/modules/tracker.js:637]: Same as above

File: src/modules/tracker.js [hunk 7]
```diff
@@ -1595,80 +1674,85 @@ class Tracker {
...
       const requestBody = applyParams(bodyParams, { ...this.options, requestMethod });
 
-      this.requests.queue(
-        requestURL,
-        requestMethod,
-        requestBody,
-        networkParameters,
-      );
-      this.requests.send();
+      if (itemId) {
```
[the remark this answers is not in the document]
Comment (esezen) [reply src/modules/tracker.js]: Good idea. Done [7861019](https://github.com/Constructor-io/constructorio-client-javascript/pull/262/commits/78610199469b8e551965bdda5ca129dc0172c2eb)

File: src/modules/tracker.js [hunk 8]
```diff
@@ -1918,29 +2017,29 @@ class Tracker {
...
-   *         quiz_session_id: '3123',
-   *         item_name: 'espresso',
-   *         variation_id: '167',
+   *         quizId: 'coffee-quiz',
+   *         quizVersionId: '1231244',
+   *         quizSessionId: '3123',
+   *         itemName: 'espresso',
+   *         variationId: '167',
    *         type: 'add_to_cart",
    *         revenue: '1.0"
```
[the remark this answers is not in the document]
Comment (esezen) [reply src/modules/tracker.js]: I don't think this happened in this PR but fixed in [7861019](https://github.com/Constructor-io/constructorio-client-javascript/pull/262/commits/78610199469b8e551965bdda5ca129dc0172c2eb)

File: spec/src/modules/tracker.js (hunk 3 above)
Comment (esezen) [reply spec/src/modules/tracker.js:799]: Good catch! I copied the initial one from Node SDK but only added `Backwards Compatibility` to the rest. Should be fixed in [0d68ed7](https://github.com/Constructor-io/constructorio-client-javascript/pull/262/commits/0d68ed7d7e04d8a7fa3e80ee372fa3766278abca)

**Comment by esezen** [comment]
@jjl014 
> Also, I noticed in some functions we specify a default section (i.e. section="Products") in the doc comments as well as defaulting the variable to Products when de-structuring. Should we do this for all functions to be consistent?

I think we should do this but do it in a separate PR. I believe some of the functions actually require section so we need to check all and update both documentation and implementation accordingly.

**Review by jjl014 — APPROVED** [review]
Nice, thanks for making the changes. This is looking great to me! 🚀 

One last thing I noticed was that there were some other places (not in this PR) that had `Backwards Compatibility ... required parameters ...` instead of `Backwards Compatibility ... snake cased parameters ...`. Should we also adjust those as well?

Other than that, feel free to ship this!

**Pull request merged**